src/federation-info-plugin/views/ServicesView/ServicesView.jsx

```diff
@@ -1,34 +1,55 @@
-import React from 'react'
+import React, { useMemo, useState, useDeferredValue } from 'react'
 import { Box } from '@mui/material'
 
 import PanelTitle from '../../components/PanelTitle/PanelTitle'
 import ServiceInfo from '../../components/ServiceInfo/ServiceInfo'
-
+import SearchServiceInput from '../../components/SearchServiceInput/SearchServiceInput'
+import filterServicesInfo from '../../utils/filterServicesInfo'
 /**
  *
  * @param {Object} props.federationServices result of prepareServicesViewData
  * @returns  {JSX.Element}
  */
-const ServicesView = ({ federationServices }) => (
-  <Box
-    sx={{
-      display: 'flex',
-      flexDirection: 'column',
-      minWidth: '200px',
-      flex: 1
-    }}
-  >
-    <PanelTitle total={federationServices.length}>Services</PanelTitle>
-    <Box sx={{ overflow: 'auto', height: '82vh', paddingRight: 2 }}>
-      {federationServices.map(({ serviceName, itemsMap }) => (
-        <ServiceInfo
-          key={serviceName}
-          serviceName={serviceName}
-          itemsMap={itemsMap}
-        />
-      ))}
+const ServicesView = ({ federationServices }) => {
+  const [query, setQuery] = useState('')
+  const defferedQuery = useDeferredValue(query)
```

Comment:
`useDeferredValue` Comes with react 18, but I think we will have to be react17 compatible due to the plugin relying on graphiql to provide react: 

https://github.com/graphql/graphiql/blob/34ade0ff2b6b9f49cc9ee81ab0427322ce41a147/packages/graphiql-react/package.json#L60